examples/contrib/detection/mean_ap.py

```diff
@@ -0,0 +1,252 @@
+"""
+Evaluation code is mainly taken credit from torchvision coco_eval.
+reference: https://github.com/pytorch/vision/blob/main/references/detection/coco_eval.py
+"""
+import copy
+import io
+from contextlib import redirect_stdout
+
+import numpy as np
+import pycocotools.mask as mask_util
+import torch
+from pycocotools import mask as coco_mask
+from pycocotools.coco import COCO
+from pycocotools.cocoeval import COCOeval
+
+from ignite.metrics import Metric
+
+
+def convert_to_coco_api(ds):
+    coco_ds = COCO()
+    # annotation IDs need to start at 1, not 0, see torchvision issue #1530
+    ann_id = 1
+    dataset = {"images": [], "categories": [], "annotations": []}
+    categories = set()
+    for img_idx in range(len(ds)):
+        # find better way to get target
+        # targets = ds.get_annotations(img_idx)
+        img, targets = ds[img_idx]
+        image_id = targets["image_id"]
+        img_dict = {}
+        img_dict["id"] = image_id
+        img_dict["height"] = img.shape[-2]
+        img_dict["width"] = img.shape[-1]
+        dataset["images"].append(img_dict)
+        bboxes = targets["boxes"].clone()
+        bboxes[:, 2:] -= bboxes[:, :2]
+        bboxes = bboxes.tolist()
+        labels = targets["labels"].tolist()
+        areas = targets["area"].tolist()
+        iscrowd = targets["iscrowd"].tolist()
+        if "masks" in targets:
```

Comment:
Here, the implementation could be simplified since no masks neither key points are used.